module/zfs/zfs_debug.c

```diff
@@ -124,7 +124,7 @@ __set_error(const char *file, const char *func, int line, int err)
 	 * $ echo 512 >/sys/module/zfs/parameters/zfs_flags
 	 */
 	if (zfs_flags & ZFS_DEBUG_SET_ERROR)
-		__dprintf(B_FALSE, file, func, line, "error %lu", err);
+		__dprintf(B_FALSE, file, func, line, "error %lu\n", err);
```

Comment:
The messages are stored without the trailing `\n` intentionally and added by `zfs_dbgmsg_print` or the proc handler.

Reply:
OK.  Last I tried it didn't print out right, but that's probably a separate issue.  Will drop the change.